l1-contracts/src/core/Decoder.sol

```diff
@@ -498,16 +498,13 @@ contract Decoder {
         calldatacopy(add(temp, 0x40), offset, iterationLogsLength)
         offset := add(offset, iterationLogsLength)
 
-        // Compute current iteration's logs hash
-        logsHash := keccak256(add(temp, 0x20), tempLength)
-
         // Decrease remaining logs length by this iteration's logs length (len(I?_LOGS)) and 4 bytes for I?_LOGS_LEN
         remainingLogsLength := sub(remainingLogsLength, add(iterationLogsLength, 0x4))
       }
 
-      // Truncate the hash to field
+      // Compute current iteration's logs hash and truncate the hash to field
       // See: https://discourse.aztec.network/t/proposal-forcing-the-sequencer-to-actually-submit-data-to-l1/426/2
-      logsHash = bytes32(uint256(logsHash) % P);
+      logsHash = bytes32(uint256(sha256(temp)) % P);
```

Comment:
You can use https://github.com/AztecProtocol/aztec-packages/blob/dd53077c7ed83a5009216aefcafe870437c30664/l1-contracts/src/core/libraries/Hash.sol#L47. I have created a small issue (#635) for updating this generally, so you don't need to handle that in here. Then we can get it in easier.